testassets/FunctionalTestsWebsite/Services/EchoService.cs

```diff
@@ -0,0 +1,120 @@
+#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Grpc.Core;
+using Grpc.Gateway.Testing;
+
+namespace FunctionalTestsWebsite.Services
+{
+    /// <summary>
+    /// Written to as closely as possible mirror the behaviour of the C++ implementation in grpc/grpc-web:
+    /// https://github.com/grpc/grpc-web/blob/92aa9f8fc8e7af4aadede52ea075dd5790a63b62/net/grpc/gateway/examples/echo/echo_service_impl.cc
```

Comment:
Seems like a good start for a set of "interop" tests, but we should reach out to grpc-web team to confirm.

Reply:
They're taken from https://github.com/johanbrandhorst/grpc-web-compatibility-test

As far as I know there are no other interop tests. We should check that with grpc-web team.